python/ray/serve/tests/test_batching.py

```diff
@@ -214,6 +215,36 @@ async def call(arg):
         await t3
 
 
+@pytest.mark.asyncio
+async def test_batch_timeout_empty_queue():
+    """Check that Serve waits for before creating the first batch."""
+
+    @serve.batch(max_batch_size=10, batch_wait_timeout_s=0.1)
+    async def no_op(requests):
+        return ["No-op"] * len(requests)
+
+    # Run one request, so internal batch queue gets constructed
+    await get_or_create_event_loop().create_task(no_op(None))
```

Comment:
this is sort of leaking an implementation detail. prefer to just run the full test twice -- that checks that this happens for the first batch and for subsequent batches in the future

Reply:
Sounds good, I made the test run twice instead of leaking the implementation detail.